pkg/render/goldmane/component.go

```diff
@@ -0,0 +1,250 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goldmane
+
+import (
+	"fmt"
+
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/ptr"
+	"github.com/tigera/operator/pkg/render"
+	"github.com/tigera/operator/pkg/render/common/securitycontext"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+)
+
+// The names of the components related to the Guardian related rendered objects.
+const (
+	GoldmaneName               = "goldmane"
+	GoldmaneNamespace          = common.CalicoNamespace
+	GoldmaneServiceAccountName = GoldmaneName
+	GoldmaneDeploymentName     = GoldmaneName
+	GoldmaneRoleName           = GoldmaneName
+	GoldmaneServicePort        = 7443
+	GoldmaneContainerName      = "goldmane"
+
+	GoldmaneKeyPairSecret = "goldmane-key-pair"
+	GoldmaneServiceName   = "goldmane"
+)
+
+func Goldmane(cfg *Configuration) render.Component {
+	c := &Component{cfg: cfg}
+
+	return c
+}
+
+// Configuration contains all the config information needed to render the component.
+type Configuration struct {
+	PullSecrets                 []*corev1.Secret
+	OpenShift                   bool
+	Installation                *operatorv1.InstallationSpec
+	TrustedCertBundle           certificatemanagement.TrustedBundleRO
+	GoldmaneServerKeyPair       certificatemanagement.KeyPairInterface
+	ManagementClusterConnection *operatorv1.ManagementClusterConnection
+	ClusterDomain               string
+}
+
+type Component struct {
+	cfg *Configuration
+
+	goldmaneImage string
+}
+
+func (c *Component) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := c.cfg.Installation.Registry
+	path := c.cfg.Installation.ImagePath
+	prefix := c.cfg.Installation.ImagePrefix
+
+	var err error
+
+	c.goldmaneImage, err = components.GetReference(components.ComponentCalicoGoldmane, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Component) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (c *Component) Objects() ([]client.Object, []client.Object) {
+	objs := []client.Object{
+		c.serviceAccount(),
+		c.role(),
+		c.roleBinding(),
+		c.goldmaneService(),
+		c.deployment(),
```

Comment:
I think we probably want a network policy here as well, and for whisker

Reply:
Are we adding net policies? I'm not opposed to it, it's just we don't have any for anything other than the apiserver.

Reply:
Yeah, in our discussion with Ben and Anthony we agreed to this.